### ISSUE TITLE
Pre-check environment variables (if required and type)

### ISSUE THREAD

**Issue by rafaelromcar-parabol** [ParabolInc/parabol#7250]
## Issue - Enhancement

Improve the error message when an environment variable isn't found or does not have the expected type.

Today, when there is an environment variable that is missing or does not have the expected type, the error message is something like the following:

```
node:buffer:322
  throw new ERR_INVALID_ARG_TYPE(
  ^

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at new NodeError (node:internal/errors:371:5)
    at Function.from (node:buffer:322:9)
    at /home/node/parabol/dist/web.js:74:1020381
    at /home/node/parabol/dist/web.js:74:1051778
    at Object.<anonymous> (/home/node/parabol/dist/web.js:74:1051782)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

It would be great to have a more detailed output, showing the name of the environment variable that does not have the correct type or is not present.

This could be done with a pre-run check of all environment variables and their types and generating an error output if some are missing or have incorrect types. Caching all the errors but still breaking the execution of the application.

### Acceptance Criteria

The application returns a more human readable and explicit message when an environment variable is missing or does not have the expected type.

**Estimated effort:** 12 points? ([see CONTRIBUTING.md](https://github.com/ParabolInc/parabol/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by rafaelromcar-parabol** [comment]
@Dschoordsch yes, sorry, it wasn't clear enough. My mistake. I'm going to edit the issue title and description.

The environment variables that are required for the application to run are not checked or if they are, the error message is not clear/explicit enough IMHO.

**Comment by jordanh** [comment]
@rafaelromcar-parabol this issue went stale (we didn't prioritize it within six months).

Feel free to reopen it and add an agenda item to the Product check-in if you'd like to prioritize. Perhaps there would be a way we could make this issue smaller (defining a pattern for args, then separate issues to tackle a few args at a time) to help make this feature easier to execute

**Comment by rafaelromcar-parabol** [comment]
Yesterday we spent some time on an issue that would have been avoided if the mandatory variables for each script were checked and validated before starting, and a clear and readable message were log.

Migrations didn't work in our newly produced docker image v6.104.0 just because it neeed two variables that wasn't present in our deployment.

This would also be very useful for any self-hosted or pubsec client that runs their own Parabol instance.

@mattkrick and I talked about he taking this subject, at least to think about it.
